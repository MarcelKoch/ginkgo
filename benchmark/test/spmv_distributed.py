#!/usr/bin/env python3
import test_framework
base_flags = ["spmv/distributed/spmv_distributed"]
# check that all input modes work:
# parameter
test_framework.compare_output_distributed(base_flags + ["-input", '[{"size": 100, "stencil": "7pt", "comm_pattern": "stencil"}]'],
                                          expected_stdout="spmv_distributed.simple.stdout",
                                          expected_stderr="spmv_distributed.simple.stderr",
                                          num_procs=3)

# stdin
test_framework.compare_output_distributed(base_flags,
                                          expected_stdout="spmv_distributed.simple.stdout",
                                          expected_stderr="spmv_distributed.simple.stderr",
                                          num_procs=3,
                                          stdin='[{"size": 100, "stencil": "7pt", "comm_pattern": "stencil"}]')

# input file
test_framework.compare_output_distributed(base_flags + ["-input", str(test_framework.sourcepath / "input.distributed_mtx.json")],
                                          expected_stdout="spmv_distributed.simple.stdout",
                                          expected_stderr="spmv_distributed.simple.stderr",
                                          num_procs=3)

# profiler annotations
test_framework.compare_output_distributed(base_flags + ["-input", '[{"size": 100, "stencil": "7pt", "comm_pattern": "stencil"}]', '-profile', '-profiler_hook', 'debug'],
                                          expected_stdout="spmv_distributed.profile.stdout",
                                          expected_stderr="spmv_distributed.profile.stderr",
                                          num_procs=3)
