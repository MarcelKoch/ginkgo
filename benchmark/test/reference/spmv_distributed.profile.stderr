This is Ginkgo 1.7.0 (develop)
    running with core module 1.7.0 (develop)
Running on reference(0)
Running with 0 warm iterations and 1 running iterations
The random seed for right hand sides is 42
The formats are [csr]x[csr]
The number of right hand sides is 1
Running test case stencil(100,7pt,stencil)
DEBUG: begin allocate
DEBUG: end   allocate
DEBUG: begin partition::build_ranges_from_global_size
DEBUG: end   partition::build_ranges_from_global_size
DEBUG: begin allocate
DEBUG: end   allocate
DEBUG: begin allocate
DEBUG: end   allocate
DEBUG: begin allocate
DEBUG: end   allocate
DEBUG: begin allocate
DEBUG: end   allocate
DEBUG: begin components::fill_array
DEBUG: end   components::fill_array
DEBUG: begin components::fill_array
DEBUG: end   components::fill_array
DEBUG: begin components::fill_array
DEBUG: end   components::fill_array
DEBUG: begin components::fill_array
DEBUG: end   components::fill_array
DEBUG: begin partition::build_from_contiguous
DEBUG: end   partition::build_from_contiguous
DEBUG: begin partition::build_starting_indices
DEBUG: end   partition::build_starting_indices
DEBUG: begin copy
DEBUG: end   copy
DEBUG: begin free
DEBUG: end   free
DEBUG: begin copy
DEBUG: end   copy
DEBUG: begin allocate
DEBUG: end   allocate
DEBUG: begin allocate
DEBUG: end   allocate
DEBUG: begin allocate
DEBUG: end   allocate
DEBUG: begin components::aos_to_soa
DEBUG: end   components::aos_to_soa
DEBUG: begin allocate
DEBUG: end   allocate
DEBUG: begin dense::fill
DEBUG: end   dense::fill
DEBUG: begin dense::fill_in_matrix_data
DEBUG: end   dense::fill_in_matrix_data
DEBUG: begin free
DEBUG: end   free
DEBUG: begin free
DEBUG: end   free
DEBUG: begin free
DEBUG: end   free
DEBUG: begin free
DEBUG: end   free
DEBUG: begin free
DEBUG: end   free
DEBUG: begin free
DEBUG: end   free
DEBUG: begin free
DEBUG: end   free
DEBUG: begin allocate
DEBUG: end   allocate
DEBUG: begin partition::build_ranges_from_global_size
DEBUG: end   partition::build_ranges_from_global_size
DEBUG: begin allocate
DEBUG: end   allocate
DEBUG: begin allocate
DEBUG: end   allocate
DEBUG: begin allocate
DEBUG: end   allocate
DEBUG: begin allocate
DEBUG: end   allocate
DEBUG: begin components::fill_array
DEBUG: end   components::fill_array
DEBUG: begin components::fill_array
DEBUG: end   components::fill_array
DEBUG: begin components::fill_array
DEBUG: end   components::fill_array
DEBUG: begin components::fill_array
DEBUG: end   components::fill_array
DEBUG: begin partition::build_from_contiguous
DEBUG: end   partition::build_from_contiguous
DEBUG: begin partition::build_starting_indices
DEBUG: end   partition::build_starting_indices
DEBUG: begin copy
DEBUG: end   copy
DEBUG: begin free
DEBUG: end   free
DEBUG: begin copy
DEBUG: end   copy
DEBUG: begin allocate
DEBUG: end   allocate
DEBUG: begin allocate
DEBUG: end   allocate
DEBUG: begin allocate
DEBUG: end   allocate
DEBUG: begin components::aos_to_soa
DEBUG: end   components::aos_to_soa
DEBUG: begin allocate
DEBUG: end   allocate
DEBUG: begin dense::fill
DEBUG: end   dense::fill
DEBUG: begin dense::fill_in_matrix_data
DEBUG: end   dense::fill_in_matrix_data
DEBUG: begin free
DEBUG: end   free
DEBUG: begin free
DEBUG: end   free
DEBUG: begin free
DEBUG: end   free
DEBUG: begin free
DEBUG: end   free
DEBUG: begin free
DEBUG: end   free
DEBUG: begin free
DEBUG: end   free
DEBUG: begin free
DEBUG: end   free
Matrix is of size (81, 81), 144
DEBUG: begin stencil(100,7pt,stencil)
	Running spmv: csr-csr
DEBUG: begin csr-csr
DEBUG: begin allocate
DEBUG: end   allocate
DEBUG: begin partition::build_ranges_from_global_size
DEBUG: end   partition::build_ranges_from_global_size
DEBUG: begin allocate
DEBUG: end   allocate
DEBUG: begin allocate
DEBUG: end   allocate
DEBUG: begin allocate
DEBUG: end   allocate
DEBUG: begin allocate
DEBUG: end   allocate
DEBUG: begin components::fill_array
DEBUG: end   components::fill_array
DEBUG: begin components::fill_array
DEBUG: end   components::fill_array
DEBUG: begin components::fill_array
DEBUG: end   components::fill_array
DEBUG: begin components::fill_array
DEBUG: end   components::fill_array
DEBUG: begin partition::build_from_contiguous
DEBUG: end   partition::build_from_contiguous
DEBUG: begin partition::build_starting_indices
DEBUG: end   partition::build_starting_indices
DEBUG: begin copy
DEBUG: end   copy
DEBUG: begin free
DEBUG: end   free
DEBUG: begin allocate
DEBUG: end   allocate
DEBUG: begin components::fill_array
DEBUG: end   components::fill_array
DEBUG: begin allocate
DEBUG: end   allocate
DEBUG: begin components::fill_array
DEBUG: end   components::fill_array
DEBUG: begin allocate
DEBUG: end   allocate
DEBUG: begin components::fill_array
DEBUG: end   components::fill_array
DEBUG: begin copy(<typename>)
DEBUG: begin copy
DEBUG: end   copy
DEBUG: begin copy
DEBUG: end   copy
DEBUG: begin copy
DEBUG: end   copy
DEBUG: begin copy
DEBUG: end   copy
DEBUG: end   copy(<typename>)
DEBUG: begin allocate
DEBUG: end   allocate
DEBUG: begin components::fill_array
DEBUG: end   components::fill_array
DEBUG: begin copy(<typename>)
DEBUG: begin copy
DEBUG: end   copy
DEBUG: begin copy
DEBUG: end   copy
DEBUG: begin copy
DEBUG: end   copy
DEBUG: begin copy
DEBUG: end   copy
DEBUG: end   copy(<typename>)
DEBUG: begin allocate
DEBUG: end   allocate
DEBUG: begin dense::fill
DEBUG: end   dense::fill
DEBUG: begin allocate
DEBUG: end   allocate
DEBUG: begin allocate
DEBUG: end   allocate
DEBUG: begin allocate
DEBUG: end   allocate
DEBUG: begin components::aos_to_soa
DEBUG: end   components::aos_to_soa
DEBUG: begin allocate
DEBUG: end   allocate
DEBUG: begin distributed_matrix::build_local_nonlocal
DEBUG: begin allocate
DEBUG: end   allocate
DEBUG: begin allocate
DEBUG: end   allocate
DEBUG: begin free
DEBUG: end   free
DEBUG: begin allocate
DEBUG: end   allocate
DEBUG: begin free
DEBUG: end   free
DEBUG: begin allocate
DEBUG: end   allocate
DEBUG: begin free
DEBUG: end   free
DEBUG: begin allocate
DEBUG: end   allocate
DEBUG: begin free
DEBUG: end   free
DEBUG: begin allocate
DEBUG: end   allocate
DEBUG: begin allocate
DEBUG: end   allocate
DEBUG: begin free
DEBUG: end   free
DEBUG: begin allocate
DEBUG: end   allocate
DEBUG: begin free
DEBUG: end   free
DEBUG: begin allocate
DEBUG: end   allocate
DEBUG: begin free
DEBUG: end   free
DEBUG: begin allocate
DEBUG: end   allocate
DEBUG: begin free
DEBUG: end   free
DEBUG: begin allocate
DEBUG: end   allocate
DEBUG: begin free
DEBUG: end   free
DEBUG: begin allocate
DEBUG: end   allocate
DEBUG: begin free
DEBUG: end   free
DEBUG: begin allocate
DEBUG: end   allocate
DEBUG: begin free
DEBUG: end   free
DEBUG: begin allocate
DEBUG: end   allocate
DEBUG: begin free
DEBUG: end   free
DEBUG: begin allocate
DEBUG: end   allocate
DEBUG: begin allocate
DEBUG: end   allocate
DEBUG: begin allocate
DEBUG: end   allocate
DEBUG: begin allocate
DEBUG: end   allocate
DEBUG: begin allocate
DEBUG: end   allocate
DEBUG: begin free
DEBUG: end   free
DEBUG: begin allocate
DEBUG: end   allocate
DEBUG: begin free
DEBUG: end   free
DEBUG: begin allocate
DEBUG: end   allocate
DEBUG: begin free
DEBUG: end   free
DEBUG: begin allocate
DEBUG: end   allocate
DEBUG: begin free
DEBUG: end   free
DEBUG: begin allocate
DEBUG: end   allocate
DEBUG: begin allocate
DEBUG: end   allocate
DEBUG: begin allocate
DEBUG: end   allocate
DEBUG: begin allocate
DEBUG: end   allocate
DEBUG: begin allocate
DEBUG: end   allocate
DEBUG: begin free
DEBUG: end   free
DEBUG: begin allocate
DEBUG: end   allocate
DEBUG: begin allocate
DEBUG: end   allocate
DEBUG: begin allocate
DEBUG: end   allocate
DEBUG: begin allocate
DEBUG: end   allocate
DEBUG: begin free
DEBUG: end   free
DEBUG: begin allocate
DEBUG: end   allocate
DEBUG: begin allocate
DEBUG: end   allocate
DEBUG: begin allocate
DEBUG: end   allocate
DEBUG: begin allocate
DEBUG: end   allocate
DEBUG: begin allocate
DEBUG: end   allocate
DEBUG: begin allocate
DEBUG: end   allocate
DEBUG: begin allocate
DEBUG: end   allocate
DEBUG: begin allocate
DEBUG: end   allocate
DEBUG: begin free
DEBUG: end   free
DEBUG: begin free
DEBUG: end   free
DEBUG: begin free
DEBUG: end   free
DEBUG: begin free
DEBUG: end   free
DEBUG: begin free
DEBUG: end   free
DEBUG: begin free
DEBUG: end   free
DEBUG: begin free
DEBUG: end   free
DEBUG: begin free
DEBUG: end   free
DEBUG: begin free
DEBUG: end   free
DEBUG: begin free
DEBUG: end   free
DEBUG: begin free
DEBUG: end   free
DEBUG: begin free
DEBUG: end   free
DEBUG: begin free
DEBUG: end   free
DEBUG: end   distributed_matrix::build_local_nonlocal
DEBUG: begin copy
DEBUG: end   copy
DEBUG: begin copy
DEBUG: end   copy
DEBUG: begin allocate
DEBUG: end   allocate
DEBUG: begin free
DEBUG: end   free
DEBUG: begin components::convert_idxs_to_ptrs
DEBUG: end   components::convert_idxs_to_ptrs
DEBUG: begin free
DEBUG: end   free
DEBUG: begin allocate
DEBUG: end   allocate
DEBUG: begin free
DEBUG: end   free
DEBUG: begin components::convert_idxs_to_ptrs
DEBUG: end   components::convert_idxs_to_ptrs
DEBUG: begin free
DEBUG: end   free
DEBUG: begin copy
DEBUG: end   copy
DEBUG: begin allocate
DEBUG: end   allocate
DEBUG: begin free
DEBUG: end   free
DEBUG: begin free
DEBUG: end   free
DEBUG: begin free
DEBUG: end   free
DEBUG: begin free
DEBUG: end   free
DEBUG: begin free
DEBUG: end   free
DEBUG: begin free
DEBUG: end   free
DEBUG: begin free
DEBUG: end   free
DEBUG: begin free
DEBUG: end   free
DEBUG: begin free
DEBUG: end   free
DEBUG: begin free
DEBUG: end   free
DEBUG: begin free
DEBUG: end   free
DEBUG: begin copy(<typename>)
DEBUG: begin allocate
DEBUG: end   allocate
DEBUG: begin dense::copy
DEBUG: end   dense::copy
DEBUG: end   copy(<typename>)
DEBUG: begin apply(<typename>)
DEBUG: begin allocate
DEBUG: end   allocate
DEBUG: begin allocate
DEBUG: end   allocate
DEBUG: begin dense::row_gather
DEBUG: end   dense::row_gather
DEBUG: begin apply(<typename>)
DEBUG: begin csr::spmv
DEBUG: end   csr::spmv
DEBUG: end   apply(<typename>)
DEBUG: begin advanced_apply(<typename>)
DEBUG: begin csr::advanced_spmv
DEBUG: end   csr::advanced_spmv
DEBUG: end   advanced_apply(<typename>)
DEBUG: end   apply(<typename>)
DEBUG: begin free
DEBUG: end   free
DEBUG: begin free
DEBUG: end   free
DEBUG: begin free
DEBUG: end   free
DEBUG: begin free
DEBUG: end   free
DEBUG: begin free
DEBUG: end   free
DEBUG: begin free
DEBUG: end   free
DEBUG: begin free
DEBUG: end   free
DEBUG: begin free
DEBUG: end   free
DEBUG: begin free
DEBUG: end   free
DEBUG: begin free
DEBUG: end   free
DEBUG: begin free
DEBUG: end   free
DEBUG: begin free
DEBUG: end   free
DEBUG: end   csr-csr
DEBUG: end   stencil(100,7pt,stencil)
DEBUG: begin free
DEBUG: end   free
DEBUG: begin free
DEBUG: end   free
